android-library/src/main/java/com/zetapush/library/ZetaPushService.kt

```diff
@@ -165,30 +160,38 @@ class ZetaPushService(
 
             Log.d("ZP", "successfulHandshake -> map = $map")
             storageTokenHandler.saveToken(map["token"] as String?)
+            listener?.successfulHandshake(map)
```

Comment:
I think java/kotlin create a final listener when you create an instance of ZPConnectionListener(), so if you change the listener instance from yout ZPService, the call of your old reference of listener.

Please add the listener in constructor and change the listener of ZPConnectionListener when you change the differents listener of your ZPService. You could do that by computed properties :).